controllers/dspipeline_controller.go

```diff
@@ -189,46 +226,133 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
+	// Initialize conditions
+	var conditions []metav1.Condition
+
+	databaseReady := r.buildCondition(config.DatabaseReady, dspa, "DatabaseReady")
 	err = r.ReconcileDatabase(ctx, dspa, params)
 	if err != nil {
+		databaseReady.Status = metav1.ConditionFalse
+		databaseReady.Reason = config.DataBaseFailedToDeploy
+		databaseReady.LastTransitionTime = metav1.Now()
```

Comment:
you have assignments to build conditions right before return statement, these won't end up getting used, same goes for every other location where there is something like:

```
something.Reason = ...
something.Status = ...
return ctrl.Result{}, err   // No status is actually updated 
```

Also, we may not want to really do anything there anyway, those error blocks are a "something gone wrong, can't proceed with reconciliation" type errors. So I don't think we need to be adding reason/statuses there.

Reply:
this was not resolved setting these values before a return statement doesn't do anything, same with here: 

https://github.com/opendatahub-io/data-science-pipelines-operator/pull/37/files#diff-e052a028f2fec91086c309e97e876f2e6491d18732fb39b270bab2b1dbe6bfcfR245-R247